infrastructure/merklemountainrange/src/merkle_change_tracker.rs

```diff
@@ -41,6 +41,8 @@ pub(crate) struct MerkleChangeTracker {
     init_key: String,
     unsaved_checkpoints: Vec<MerkleCheckPoint>,
     uncleaned_checkpoints: Vec<CheckpointCleanup>,
+    unpruned_indexes: Treemap,
+    pruned_indexes: Treemap,
```

Comment:
From what I can tell, this field is mainly used to reset `unpruned_indexes` (note: plural of index = indices) when doing a rewind. Do you feel that tracking the pruned indices in a RB is better than bundling the leaf index in the checkpoint data?

Reply:
Fixed the spelling mistake

but yes I feel it will be better for two reasons:
-  My suspicion is that those indices could get quite large with large blocks
- There is nice clean code for merging/deleting between two RB's, using individual indices would result in messier code.